### PR TITLE
Add support for unicode string inputs to Workflow Transform in Triton

### PR DESCRIPTION
**Pull request by oliverholworthy**
Add support for unicode string inputs to Workflow Transform in Triton

### CODE REVIEW

File: merlin/systems/triton/__init__.py [hunk 1]
```diff
@@ -150,9 +150,6 @@ def _convert_tensor(t):
     out = t.as_numpy()
     if len(out.shape) == 2:
         out = out[:, 0]
```
Comment (oliverholworthy) [inline merlin/systems/triton/__init__.py:152]: Unrelated to this change: It's unclear to me why we'd want to remove dimensions from the input here

File: merlin/systems/triton/__init__.py [hunk 2]
```diff
@@ -150,9 +150,6 @@ def _convert_tensor(t):
...
-    # cudf doesn't seem to handle dtypes like |S15 or object that well
-    if is_string_dtype(out.dtype):
-        out = out.astype("str")
```
[the remark this answers is not in the document]
Comment (karlhigley) [reply merlin/systems/triton/__init__.py:155]: My impression is that this does (or did) cover a real edge case, which may not be adequately covered by tests. This piece of code was inherited from the old serving code in NVT, which was TBH not very well tested.

File: merlin/systems/triton/__init__.py (hunk 1 above)
Comment (karlhigley) [reply merlin/systems/triton/__init__.py:152]: This code has existed for a long time, and I think is related to the perennial inconsistency around list formats that has plagued the Merlin code base. Way back in the before times, sometimes you'd get a proper 1d array/tensor and sometimes you'd get a 2d array/tensor that only contained one row. The legacy serving code from NVT that Systems is based on and still trying to clean up and/or shed had all kinds of issues like this and mostly solved them by hacking around the inconsistent formats instead of standardizing.

File: merlin/systems/triton/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (oliverholworthy) [reply merlin/systems/triton/__init__.py:155]: I've updated this keeing the string type coercion. Using `np.char.decode(out.astype(bytes))` intead of `out.astype("str")`.

It appears we do need this because cudf doesn't accept an array of byte strings as a type when constructing a DataFrame.
